### DISCUSSION TITLE
VFS  handle custom ioctl?

### DISCUSSION THREAD

**Discussion by tdgvrr** [dCache/nfs4j#102] [Q&A] [answered]
Trying to understand the VFS implementation better...is there a way I can have a virtual file system that can receive various control functions from a client through some sort of ioctl (or equivalent)? 

We have a virtual filesystem implemented as a database and we'd like to expose certain functions to the client ("COMMIT" and "ROLLBACK" for example)...is this sort of thing possible in the structure you guys provide?

Thanks!

**Comment by kofemann** [comment, marked as answer]
Hi,

First of all thanks for looking at nfs4j.

The nfs protocol doesn't describe any possibilities for a client to send a command that is not defined by the standard.

Actually, in dCache indeed we need a similar functionality and have implemented it as *magic files*, something like :

```
touch .(<command>)(<file>) 
```

There are of course some issues with this.

- it is limited by max file name (255)
- hard to get clients cache to play with invisible files

However, a similar solution can be adopted for your needs.
